### DISCUSSION TITLE
Question on Customizing `Slider` Component Theme for Disabled State

### DISCUSSION THREAD

**Discussion by guilean** [chakra-ui/chakra-ui#8405] [Q&A] [answered]
Hi there!

Firstly, I want to express my appreciation for all the work that goes into Chakra UI—it's been a fantastic tool for our projects. 🤗

I'm currently working on customizing the `Slider` component, specifically trying to apply a style when the component is in a disabled state using the `isDisabled` property.

I've noticed that for parts of the `Slider` such as the `thumb`, `track`, and the `container`, there are `data-disabled` attributes or `aria-disabled` which are really handy for applying the `_disabled` selector in the theme. However, this doesn't seem to be the case for the `filledTrack` part. It looks like the `filledTrack` doesn't receive any `data-disabled` attribute in the DOM, making it challenging to apply disabled styles consistently with the other parts.

Is this behavior by design, or would it be expected for the `filledTrack` part to also indicate its disabled state with an attribute, similar to the other components of the `Slider`? Understanding this will help me determine whether I should look for an alternative styling approach or consider working on a proposal for this feature if it's an oversight or unexpected behavior.

Thanks so much for your time and assistance.

**Comment by segunadebayo** [comment, marked as answer]
`isDisabled` and the old Slider parts are v2. v2 is maintenance-only.

In v3, style `range` from the root `_disabled` slot. The thumb and track already get `data-disabled`.
